plugins/alibaba/ram/inactiveUserDisabled.js

```diff
@@ -0,0 +1,65 @@
+var helpers = require('../../../helpers/alibaba');
+
+module.exports = {
+    title: 'Inactive User Disabled',
+    category: 'RAM',
+    description: 'Ensures a ram user inactive for 90 or more days is disabled',
+    more_info: 'RAM User should not have the console access enabled on being inactive for 90 or more days.',
+    link: 'https://alibaba-cloud.medium.com/11-security-recommendations-for-production-instances-on-alibaba-cloud-960e3e8442d4',
+    recommended_action: 'Disable a ram user if its inactive for 90 or more days',
+    apis: ['RAM:ListUsers', 'RAM:GetUser', 'RAM:GetLoginProfile', 'STS:GetCallerIdentity'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+
+        var region = helpers.defaultRegion(settings);
+        var accountId = helpers.addSource(cache, source, ['sts', 'GetCallerIdentity', region, 'data']);
+        var listUsers = helpers.addSource(cache, source, 
+            ['ram', 'ListUsers', region]);
+
+        if (!listUsers) return callback(null, results, source);
+
+        if (listUsers.err || !listUsers.data) {
+            helpers.addResult(results, 3,
+                'Unable to query RAM users' + helpers.addError(listUsers), region);
+            return callback(null, results, source);
+        }
+
+        if (!listUsers.data.length) {
+            helpers.addResult(results, 0, 'No RAM users found', region);
+            return callback(null, results, source);
+        }
+
+        for (var user of listUsers.data) {
+            if (!user.UserName) continue;
+            
+            var getUser = helpers.addSource(cache, source,
+                ['ram', 'GetUser', region, user.UserName]);
+            
+            let lastLoginDate = getUser.data.LastLoginDate.length ? getUser.data.LastLoginDate:getUser.data.CreateDate ;
+            var currentDate = new Date();
+            var loginDate = new Date(lastLoginDate);
+            var resource = helpers.createArn('ram', accountId, 'user', user.UserName);
+
+            var diffInDays = helpers.daysBetween(currentDate, loginDate);
+            if (diffInDays >= 90) {
+                var getUserProfile = helpers.addSource(cache, source,
+                    ['ram', 'GetLoginProfile', region, user.UserName]);
+
+                if (getUserProfile.data.LoginProfile) {
+                    helpers.addResult(results, 2,
+                        'RAM user inactive for 90 or more days is enabled ', region, resource);
+                } else {
+                    helpers.addResult(results, 0,
+                        'RAM user inactive for 90 or more days is not enabled', region, resource);
+                }
+            } else {
+                helpers.addResult(results, 0,
+                    'RAM user inactive for 90 or more days is not enabled', region, resource);
```

Comment:
here it is checked that if the diffdays is not greater or equal 90, but we are not checking the login profile, But in message we are saying user is not enabled. as our check is only for user with days limit >=90 here i think we should not have a else at all as we are not interested there, or we need to check the user profile as well before saying its not enabled. what if some user has kept enabled but last access is less than 90. it will give result as  as not enabled.